src/main/target/STM32H750/target.h

```diff
@@ -18,16 +18,51 @@
  * If not, see <http://www.gnu.org/licenses/>.
  */
 
+/*
+ * This target won't actually build a target that you can boot, it is meant as
+ * a base target ONLY.
+ *
+ * When defining a target that uses this as a base you currently:
+ *
+ * 1) *must* define ALL the SPI instances that the target has, including their pins.
+ * 2) *must* define the storage subsystem used to boot, e.g. see CONFIG_IN_xxx.
+ * 3) *must* define all the settings required for the config storage system to
+ *    be able to load the config at boot time BEFORE it has actually loaded a config.
+ *    e.g. for QSPI define the QSPI instance, pins, mode, etc,
+ *         for SDCARD define the SD card bus (SPI/SDIO), pins, etc.
+ *
+ */
+
 #pragma once
 
+#ifndef TARGET_BOARD_IDENTIFIER
 #define TARGET_BOARD_IDENTIFIER "S750"
+#endif
 
+#ifndef USBD_PRODUCT_STRING
 #define USBD_PRODUCT_STRING     "Betaflight STM32H750"
+#endif
 
+#if !defined(USE_I2C)
```

Comment:
For this one the point is not to define I2C instances targets that do not have the pins wired, this is especially important for the cloud build system where you need to save space.  remember that the H750 only has 448K of RAM to use for firmware.

Similarly it's pretty pointless compiling in the MAX7456 driver and SD CARD SPI by default in `target/common_pre.h` when the board you're using, respectively, either has NO OSD (MSP Display Port use only); FrSKY OSD onboard or SPRacingPixelOSD onboard, or SD CARD using SDIO.  There's no need for all the code bloat for drivers that the target physically cannot and will not ever have.

To me it feels like we almost need the opposite of `CLOUD_BUILD`, but as a specific define, that we can use, e.g. `BASE_TARGET_BUILD` or `UNIFIED_BUILD`.  Then `unified` targets will not have the bloat of the `BASE_TARGET_BUILD` and can be combined with `CLOUD_BUILD` so that required stuff IS included and code/defines required for `BASE_TARGETS` still has a place to live.

maybe 'complete' is a better term than 'unified' now...

I dunno, thinking out loud, but the build scenarios are:
1) build a 'complete mcu target' for large flash non-EXST targets, i.e. the 'previously-known-as-unified-target' binaries.
2) build a 'base mcu target' (includes 100% necessary code only) to which a 'unified target config' is applied which contains per-target conditional defines, but that still builds /without/ a target config from the target configs repo applied to it, but that doesn't necessarily actually boot, aka the current STM32H750/STM32H730 targets.
3) build a 'target' (again, includes only 100% necessary code without all the 'complete target' bloat, and based on a specific 'base mcu target', e.g. the STM32H750 'base mcu target' with the SPRacingH7EXTREME defines applied.
4) build a 'cloud build target' (again, includes only 100% necessary code without all the 'complete target' bloat, and based on a specific 'base mcu target'), e.g. some 'base mcu target' with just what the user wants added.

Reply:
As examples: MAX7456 and SD CARD don't get added in - unless you request a classic build OR they are present in the target configuration files.

See: https://github.com/betaflight/unified-targets/blob/master/configs/default/HAMO-CRAZYBEEF4SX1280.config as an example.

CLOUD_BUILD explicitly excludes these items in `common_pre.h` this is by design. We will end up adding another flag that is CLASSIC that will add in all the applicable USER options (if cloud build), but still utilise only the #defines in the target configuration.

Reply:
I'm not sure your statement is correct, or I'm mis-understanding something, but unless I'm failing to read code i see that both USE_MAX7456 and USE_SDCARD are both #defined in this block:

https://github.com/betaflight/betaflight/blob/master/src/main/target/common_pre.h#L236-L294

the gating for the block is:

```
#if !defined(CLOUD_BUILD) && !defined(SITL)
```

As noted in another comment recently the SPRacingH7EXTREME 'unified' target config should NOT specify 'CLOUD_BUILD' in the `.config` file and thus all the #defines in the block above WILL be included in the target as `!CLOUD_BUILD && !SITL == true`.

Or is there some other magic code somewhere doing something I'm not aware of?

I'm assuming:
* classic mode = ON, don't add the 'CLOUD_BUILD'.
* classic mode = OFF, add 'CLOUD_BUILD'.

which IMHO is a bad term as it doesn't tell the user anything, especially if they've never encountered BF or it's build system settings before, it should probably be called 'COMPLETE' using the terminology from this comment: https://github.com/betaflight/betaflight/pull/12091#discussion_r1057367494

As per that same comment some build scenarios seem to be missing from the cloud build system, configurator ui, documentation and codebase.

Reply:
**classic mode = ON** 
don't add the 'CLOUD_BUILD', and only include the board_name, manufacturer_id and build_key defines. 
NOTE: In RC3 I will look to be adding in all the defines included in the config file also, so it will not include a lot of the "drivers" if they are provided in the config file. 

**classic mode = OFF**
add 'CLOUD_BUILD', any defines from the config file, and the user selection defines.